src/days-2-4-adv/adv.py

```diff
@@ -49,3 +50,115 @@
+#                 suppressRoomPrint = True
+# #IF THERE IS ONLY 1 INPUT AND IT IS ONE OF THESE LETTERS
+# #IT WILL SET THE CURRENTROOM ATTR TO DIRECTION
+# #OR ELSE IT WILL PRINT THE YOU CANNOT MSG
+#         elif cmds[0] == 's':
+#             if player.currentRoom.s_to is not None:
+#                 player.currentRoom = player.currentRoom.s_to
+#             else:
+#                 print("You cannot move in that direction")
+#                 suppressRoomPrint = True
+#         elif cmds[0] == 'w':
+#             if player.currentRoom.w_to is not None:
+#                 player.currentRoom = player.currentRoom.w_to
+#             else:
+#                 print("You cannot move in that direction")
+#                 suppressRoomPrint = True
+#         elif cmds[0] == 'e':
+#             if player.currentRoom.e_to is not None:
+#                 player.currentRoom = player.currentRoom.e_to
+#             else:
+#                 print("You cannot move in that direction")
+#                 suppressRoomPrint = True
+#         else:
+#             print("need valid command")
+#             suppressRoomPrint = True
+#     else:
+#         if cmds[0] == "look":
+#             if cmds[1] == "n":
+#                 if player.currentRoom.n_to is not None:
+#                     print(f"\nin '{cmds[1]}' direction:", player.currentRoom.n_to)
+#                     suppressRoomPrint = True
+#                 else:
+#                     print("There is nothing to look at")
+#                     suppressRoomPrint = True
+#             elif cmds[1] == "s":
+#                 if player.currentRoom.s_to is not None:
+#                     print(f"\nin '{cmds[1]}' direction:", player.currentRoom.s_to)
+#                     suppressRoomPrint = True
+#                 else:
+#                     print("There is nothing to look at")
+#                     suppressRoomPrint = True
+#             elif cmds[1] == "w":
+#                 if player.currentRoom.w_to is not None:
+#                     print(f"\nin '{cmds[1]}' direction:", player.currentRoom.w_to)
+#                     suppressRoomPrint = True
+#                 else:
+#                     print("There is nothing to look at")
+#                     suppressRoomPrint = True
+#             elif cmds[1] == "e":
+#                 if player.currentRoom.e_to is not None:
+#                     print(f"\nin '{cmds[1]}' direction:", player.currentRoom.e_to)
+#                     suppressRoomPrint = True
+#                 else:
+#                     print("There is nothing to look at")
+#                     suppressRoomPrint = True
+#         else:
+#             print("need valid commands")
+#             suppressRoomPrint = True
+
+
+#2ND IMPLEMENTATION
+valid_directions = {"n": "n", "s": "s", "e": "e", "w": "w",
+                    "north": "n", "south": "s", "east": "e", "west": "w",
+                    "forward": "n", "backward": "s", "right": "e", "left": "w"}
+
```

Comment:
Interesting that you put keys for forward/backward, left/right and not just nsew.  Makes sense since not everyone thinks about movement in terms of compass directions.